# 데이터 다 썼을때

In [1]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn import svm
from sklearn import metrics

In [2]:
df = pd.read_csv('titanic.csv')
data_df = df
data_df.head()

pclass                               name  sex   age  sibsp  parch  \
0       2  Mellinger, Miss. Madeleine Violet    0  13.0      0      1   
1       2                  Wells, Miss. Joan    0   4.0      1      1   
2       2     Duran y More, Miss. Florentina    0  30.0      1      0   
3       3       Bradley, Miss. Bridget Delia    0  22.0      0      0   
4       1     Francatelli, Miss. Laura Mabel    0  30.0      0      0   

          ticket     fare cabin embarked  survived  
0         250644  19.5000   NaN        S         1  
1          29103  23.0000   NaN        S         1  
2  SC/PARIS 2148  13.8583   NaN        C         1  
3         334914   7.7250   NaN        Q         1  
4       PC 17485  56.9292   E36        C         1

In [3]:
# 데이터 쪼개기
train, test = train_test_split(data_df, test_size=0.2, random_state=42)
print('train data:',train.shape)
print('test data:',test.shape)

train data: (836, 11)
test data: (210, 11)


In [4]:
data_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1046 entries, 0 to 1045
Data columns (total 11 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   pclass    1046 non-null   int64  
 1   name      1046 non-null   object 
 2   sex       1046 non-null   int64  
 3   age       1046 non-null   float64
 4   sibsp     1046 non-null   int64  
 5   parch     1046 non-null   int64  
 6   ticket    1046 non-null   object 
 7   fare      1046 non-null   float64
 8   cabin     272 non-null    object 
 9   embarked  1044 non-null   object 
 10  survived  1046 non-null   int64  
dtypes: float64(2), int64(5), object(4)
memory usage: 90.0+ KB


In [5]:
data_df.corr(numeric_only=True)

pclass       sex       age     sibsp     parch      fare  survived
pclass    1.000000  0.144695 -0.408106  0.047221  0.017224 -0.565203 -0.320486
sex       0.144695  1.000000  0.063645 -0.095267 -0.221144 -0.188036 -0.538000
age      -0.408106  0.063645  1.000000 -0.243699 -0.150917  0.177895 -0.055512
sibsp     0.047221 -0.095267 -0.243699  1.000000  0.374456  0.141278 -0.012213
parch     0.017224 -0.221144 -0.150917  0.374456  1.000000  0.216799  0.114438
fare     -0.565203 -0.188036  0.177895  0.141278  0.216799  1.000000  0.249253
survived -0.320486 -0.538000 -0.055512 -0.012213  0.114438  0.249253  1.000000

In [6]:
train_data_df = train[['pclass','sex']]
train_data_df.head()

pclass  sex
256       2    1
351       2    1
60        2    0
344       3    0
381       2    1

In [7]:
train_label_df = train[['survived']]
train_label_df.head()

survived
256         0
351         0
60          1
344         1
381         0

In [8]:
train_data = train_data_df.values  
train_data

array([[2, 1],
       [2, 1],
       [2, 0],
       ...,
       [3, 0],
       [2, 0],
       [3, 1]])

In [9]:
train_label = train_label_df.values.ravel()
train_label

array([0, 0, 1, 1, 0, 1, 0, 1, 1, 0, 0, 1, 0, 1, 0, 1, 0, 0, 0, 0, 1, 1,
       1, 0, 0, 0, 1, 1, 0, 0, 1, 0, 1, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0,
       1, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0,
       0, 0, 0, 0, 1, 0, 1, 0, 1, 0, 0, 0, 1, 0, 0, 1, 0, 1, 0, 0, 0, 1,
       1, 0, 1, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 1, 0, 0,
       0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 1, 1, 0, 0, 0, 1, 0, 0, 1, 0, 1,
       0, 0, 1, 1, 0, 1, 1, 1, 0, 0, 1, 0, 1, 0, 0, 1, 0, 0, 1, 1, 0, 0,
       0, 0, 1, 0, 0, 0, 1, 0, 0, 1, 1, 1, 0, 0, 1, 0, 0, 1, 0, 0, 0, 1,
       0, 0, 1, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0,
       1, 0, 1, 0, 0, 1, 0, 1, 0, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 0, 1,
       0, 1, 1, 1, 0, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 1, 0, 0,
       0, 0, 0, 1, 1, 0, 1, 1, 0, 0, 1, 1, 0, 0, 1, 1, 0, 0, 0, 1, 1, 1,
       0, 0, 1, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 1, 0, 0,
       1, 0, 0, 1, 1, 0, 1, 1, 0, 0, 0, 0, 0, 1, 0,

In [10]:
test_data_df = test[['pclass','sex']]
test_label_df = test[['survived']]
test_data = test_data_df.values
test_label = test_label_df.values.ravel()

In [11]:
clf = svm.SVC(C=1, gamma= 0.1)
clf.fit(train_data, train_label)

SVC(C=1, gamma=0.1)

In [12]:
pred_svm = clf.predict(test_data)
pred_svm

array([0, 1, 0, 0, 0, 1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 1, 0,
       1, 0, 1, 0, 1, 0, 0, 1, 1, 1, 0, 0, 0, 0, 1, 0, 0, 1, 0, 1, 0, 0,
       1, 0, 1, 0, 1, 0, 0, 0, 1, 0, 1, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 1,
       0, 1, 0, 1, 0, 0, 0, 0, 1, 1, 1, 0, 1, 1, 1, 0, 0, 0, 1, 0, 0, 0,
       0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0,
       1, 1, 1, 0, 1, 1, 0, 1, 0, 0, 0, 1, 1, 0, 1, 0, 1, 0, 1, 1, 0, 1,
       0, 0, 0, 1, 1, 0, 0, 1, 0, 1, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0, 1,
       1, 0, 1, 1, 1, 0, 1, 1, 0, 0, 0, 1, 1, 1, 0, 0, 1, 0, 0, 0, 0, 1,
       1, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 1, 1, 0, 0, 0,
       1, 0, 0, 0, 0, 1, 0, 0, 1, 0, 1, 1])

In [13]:
ac_score= metrics.accuracy_score(test_label, pred_svm)
print('정확도:', ac_score)

정확도: 0.7476190476190476


In [14]:
comparasion_df = pd.DataFrame({'prediction_df': pred_svm , 'ground_truth': test_label})
comparasion_df

prediction_df  ground_truth
0                0             0
1                1             1
2                0             1
3                0             0
4                0             1
..             ...           ...
205              0             0
206              1             0
207              0             0
208              1             1
209              1             0

[210 rows x 2 columns]

# 데이터 전부 사용, 피쳐 변경
* 정확도 : 74.76%
* c : 1 , gamma:0.1
* 피쳐 [['pclass','sex']]